frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.boot-order.scenario.ts

```diff
@@ -0,0 +1,137 @@
+  const testVM = getVMManifest('Container', testName, `bootordervm-${getRandStr(5)}`);
+  const vm = new VirtualMachine(testVM.metadata);
+
+  beforeAll(() => {
+    createResource(testVM);
+  });
+
+  afterAll(() => {
+    deleteResource(testVM);
+  });
+
+  beforeEach(async () => {
+    await vm.navigateToDetail();
+    await vm.modalEditBootOrder();
+  });
+
+  it(
+    'Displays boot devices',
+    async () => {
+      const bootableDevices = getBootableDevices(getResourceObject(vm.name, vm.namespace, vm.kind));
+      const displayedbootableDevices = await vm.getBootDevices();
+      expect(displayedbootableDevices.length).toEqual(bootableDevices.length);
+    },
+    VM_CREATE_AND_EDIT_TIMEOUT_SECS,
+  );
+
+  it(
+    'Deletes bootable device',
+    async () => {
+      const FIRST_DEVICE_POSITION = 1;
+      const initialBootableDevices = getBootableDevicesInOrder(
+        getResourceObject(vm.name, vm.namespace, vm.kind),
+      );
+      await click(bootOrderView.deleteDeviceButton(FIRST_DEVICE_POSITION));
+      await click(bootOrderView.saveButton);
+      // Wait for the boot Order to update
+      await browser.wait(
+        waitForCount(
+          bootOrderView.bootableDevices(vm.namespace, vm.name),
+          initialBootableDevices.length - 1,
+        ),
+      );
+      const orderedBootableDevices = getBootableDevicesInOrder(
+        getResourceObject(vm.name, vm.namespace, vm.kind),
+      ).map((device) => `${_.get(device, 'value.name')}`);
+      const displayedbootableDevices = (await vm.getBootDevices()).map(
+        (device) => device.split(' ')[0],
+      );
+      expect(orderedBootableDevices).toEqual(displayedbootableDevices);
+    },
+    VM_CREATE_AND_EDIT_TIMEOUT_SECS,
+  );
+
+  it(
+    'Adds bootable device',
+    async () => {
+      const initialBootableDevices = getBootableDevicesInOrder(
+        getResourceObject(vm.name, vm.namespace, vm.kind),
+      );
+      await click(bootOrderView.addDeviceButton);
+      const bootableDevices = await getSelectOptions(bootOrderView.addDeviceSelect);
```

Comment:
can we check that the options contain all non bootable devices?

Reply:
Done